verifiers/utils/sigstore_tuf.go

```diff
@@ -0,0 +1,64 @@
+package utils
+
+import (
+	"sync"
+
+	sigstoreRoot "github.com/sigstore/sigstore-go/pkg/root"
+	sigstoreTUF "github.com/sigstore/sigstore-go/pkg/tuf"
+)
+
+var (
+	// cache the default Sigstore TUF client.
+	defaultSigstoreTUFClient *sigstoreTUF.Client
+	// defaultSigstoreTUFClientOnce is used for initializing the defaultSigstoreTUFClient.
+	defaultSigstoreTUFClientOnce = new(sync.Once)
+
+	// cache the trusted root.
+	trustedRoot *sigstoreRoot.TrustedRoot
+	// trustedRootOnce is used for initializing the trustedRoot.
+	trustedRootOnce = new(sync.Once)
+)
+
+// SigstoreTUFClient is the interface for the Sigstore TUF client.
+type SigstoreTUFClient interface {
+	// GetTarget retrieves the target file from the TUF repository.
+	GetTarget(target string) ([]byte, error)
+}
+
+// GetDefaultSigstoreTUFClient returns the default Sigstore TUF client.
+// The client will be cached in memory.
+func GetDefaultSigstoreTUFClient() (*sigstoreTUF.Client, error) {
+	var err error
+	defaultSigstoreTUFClientOnce.Do(func() {
+		defaultSigstoreTUFClient, err = sigstoreTUF.DefaultClient()
+		if err != nil {
+			defaultSigstoreTUFClientOnce = new(sync.Once)
+			return
+		}
+	})
+	if err != nil {
+		return nil, err
+	}
+	return defaultSigstoreTUFClient, nil
+}
+
+// GetSigstoreTrustedRoot returns the trusted root for the Sigstore TUF client.
+func GetSigstoreTrustedRoot() (*sigstoreRoot.TrustedRoot, error) {
+	var err error
+	trustedRootOnce.Do(func() {
+		client, err := GetDefaultSigstoreTUFClient()
+		if err != nil {
+			trustedRootOnce = new(sync.Once)
+			return
+		}
+		trustedRoot, err = sigstoreRoot.GetTrustedRoot(client)
```

Comment:
Should we handle slsa-verifier used in any long-running (> 1 day) processes? If so, then we should look at using [LiveTrustedRoot](https://github.com/sigstore/sigstore-go/blob/2b6fc6df45dc26c329f4da6c8157340e0f435a0d/pkg/root/trusted_root.go#L310) to refresh the root periodically.

Reply:
It makes sense to do that, but I feel we need to change the `NewLiveTrustedRoot` to accept an existing TUFClient, rather than make a new one. wdyt?

- https://github.com/sigstore/sigstore-go/blob/2b6fc6df45dc26c329f4da6c8157340e0f435a0d/pkg/root/trusted_root.go#L310

from 

```
func NewLiveTrustedRoot(opts *tuf.Options) (*LiveTrustedRoot, error)
```

to

```
func NewLiveTrustedRoot(c *tuf.Client) (*LiveTrustedRoot, error)
```

Reply:
IIUC, we don't need to explicitly cache TrustedRoot, because the TUFClient will already cache it with GetTarget(). Is that right?

Reply:
following up in https://github.com/sigstore/sigstore-go/issues/249

Reply:
`GetTarget` caches the target file to prevent redownloading it. Separately, the TUF client will still check if the local TUF repository metadata needs to be updated. If caching is configured (https://github.com/sigstore/sigstore-go/blob/main/pkg/tuf/options.go#L47), then the client will only fetch new targets once the local timestamp has expired or the cache validity window has passed.

For an API to accept a TUF client, one issue is that the underlying Updater can only have Refresh called once (https://github.com/theupdateframework/go-tuf/blob/f95222bdd22d2ac4e5b8ed6fe912b645e213c3b5/metadata/updater/updater.go#L105) (I'm not sure why that is the decision, we can ask Fredrik on the issue). If the live updater accepts an existing TUF client, it will need to reinitialize the `Updater`, which will need the options.

Reply:
Since the current implementation just supports verifying from the public instance, WDYT about proceeding with using `NewLiveTrustedRoot` as-is, since a custom TUF client shouldn't be necessary?

Reply:
Added. 

It seems counter to our other plans for allowing slsa-verifier users to pass their own TUF client. But slsa-verifier needs to do live lookups against the public Sigstore infra, yes, I would be okay with not using the user-provided TUF client.

- https://github.com/slsa-framework/slsa-verifier/pull/768